deviations/views.py

```diff
@@ -95,3 +148,48 @@ def get_resource_objects(self):
     def post(self, request, *args, **kwargs):
         self.deviation.delete()
         return self.redirect(self.instance.get_url("deviations-list-dl"))
+
+
+class RemoveManyDeadlinesView(CourseInstanceMixin, BaseFormView):
+    access_mode = ACCESS.TEACHER
+    template_name = "deviations/remove_dl.html"
+    form_class = RemoveDeadlineRuleDeviationForm
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["instance"] = self.instance
+        return kwargs
+
+    def get_success_url(self):
+        return self.instance.get_url("deviations-list-dl")
+
+    def form_valid(self, form):
+        number_of_removed = 0
+        for profile in form.cleaned_data["submitter"]:
+            deviation_exercises = form.cleaned_data["exercise"]
+            for module in form.cleaned_data["module"]:
+                exercises = BaseExercise.objects.filter(course_module=module)
```

Comment:
I think that the exercises should be queried from the database outside the `for profile` loop. The exercises are the same for all users so you do not need to query the exercises repeatedly for each user.